app/workers/job_concurrency.rb

```diff
@@ -36,7 +36,24 @@ module JobConcurrency
   ##
   # Run the concurrency check of good_job without actually trying to enqueue it
   # Will call the provided block in case the job would be cancelled
-  def check_concurrency(&block)
-    good_job_enqueue_concurrency_check(self, on_abort: block, on_enqueue: nil)
+  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable Metrics/PerceivedComplexity
+  def check_concurrency
+    enqueue_limit = self.class.good_job_concurrency_config[:enqueue_limit]
+    enqueue_limit = instance_exec(&enqueue_limit) if enqueue_limit.respond_to?(:call)
+    enqueue_limit = nil unless enqueue_limit.present? && (0...Float::INFINITY).cover?(enqueue_limit)
+
+    unless enqueue_limit
+      total_limit = self.class.good_job_concurrency_config[:total_limit]
+      total_limit = instance_exec(&total_limit) if total_limit.respond_to?(:call)
+      total_limit = nil unless total_limit.present? && (0...Float::INFINITY).cover?(total_limit)
```

Comment:
:green_circle: Readability question: Is there a reason why this is not phrased as

```suggestion
      total_limit = nil unless total_limit.present? && total_limit >= 0
```

or even

```suggestion
      total_limit = nil if total_limit.blank? || total_limit < 0
```

Reply:
Maybe the answer is that we want to set the limit to `nil`, if `Float::INFINITY` was passed, because the range is not `0..Float::INFINITY`, but `0...Float::INFINITY`, so it does not include infinity itself.

That said, I am not sure if it would be a bad thing to set the limit to infinity...